SCClassLibrary/Common/Collections/SequenceableCollection.sc

```diff
@@ -314,7 +314,28 @@ SequenceableCollection : Collection {
 	top { ^this.last }
 	putFirst { arg obj; if (this.size > 0, { ^this.put(0, obj) }) }
 	putLast { arg obj; if (this.size > 0, { ^this.put(this.size - 1, obj) }) }
+	insertAll { arg index, obj;
+		^this[..index-1] ++ obj ++ this[index..]
+	}
+
+	mergeItems { |aCollection| // can be parametrized in the future
+		var i, j, coll, rest;
```

Comment:
these should be declared in the `else` branch below, not here

Reply:
no, not so good, that would prevent inlining. It would also post inline warnings to inform, if switched on.

Reply:
Telephon is correct. The main class library should inline control structures whenever possible. SC != C++.

I've said before, imo all devs should switch on the inline warnings. Dev code should be held to a tighter standard on this particular matter.

Reply:
> SC != C++.

Not just a C++ thing. 'declare variables in the smallest available scope' is good style in most languages. Sucks that sc makes good code hard to write, but I'm also struggling to understand what the actual performance impact. Point me to some documentation if you know where it is, i've been looking for awhile in GH issues and scdoc.

Reply:
I'll answer the inlining question on the mailing list.

Reply:
thank you, @jamshark70. Good explanation, and I would be for keeping inline warnings `true` for all builds, not just for developers. It just makes coding styles diverge if we keep in off for longer.

I also would favor a lispier style in such cases. But in the end, if the declarations are too far away, the code can probably be refactored in some other way. But I don't see that problem here in this case.